modules/api/php/module.class.inc

```diff
@@ -106,24 +108,43 @@ class Module extends \Module
             ->withAttribute("LORIS-API-Version", $version)
             ->withAttribute('pathparts', $pathparts);
 
-        $parentresp = parent::handle($newrequest);
-        switch ($parentresp->getStatusCode()) {
+        $handlername = $pathparts[0];
+        switch($handlername) {
+        case 'login':
+            $handler = new \LORIS\Api\Endpoints\Login();
+            break;
+        case 'projects':
+            $handler = new \LORIS\Api\Endpoints\Projects();
+            break;
+        default:
+            return (new \LORIS\Http\Response())
+                    ->withStatus(404)
+                    ->withHeader('Content-Type', 'application/json')
+                    ->withBody(
+                        new \LORIS\Http\StringStream(
+                            json_encode(["error" => "not found"])
+                        )
+                    );
+        }
+
+        $response = $handler->handle($newrequest);
```

Comment:
```suggestion
        // This disregards the parent handler from the base module class, because most of the work
        // it does (adding page decoration middleware, delegating to a sub-page class in the Module, etc)
        // is not relevant for the API module.
        $response = $handler->handle($newrequest);
```